### DISCUSSION TITLE
Texifier打开模板出现乱码并无法编译

### DISCUSSION THREAD

**Discussion by Larerr** [BITNP/BIThesis#433] [Q&A]
### 发生了什么？

Texifier打开模板出现乱码并无法编译QAQ，请问怎么解决呀
![texifier](https://github.com/BITNP/BIThesis/assets/101967188/2ed9cd65-1d88-409b-a00a-4749291f9549)

### 问题复现的步骤

1.打开texifier
2.打开main.tex文件

### 最小可复现用例或者截图

![texifier](https://github.com/BITNP/BIThesis/assets/101967188/f8a46570-fffa-441d-a38a-e1dda2a89ee8)

### 操作系统或平台

None

**Comment by YDX-2147483647** [comment]
模板文件的编码是UTF-8（XeLaTeX应该也要求用UTF-8），Texifier乱码可能是因为用别的编码读了。（Texifier具体怎么设置要查查……）

### 更新

按照 [Texifier · Docs · File encoding: typesetting accents and other non-ASCII characters with Texifier](https://www.texifier.com/docs/apps/misc/languages/encoding) 的说法，它默认也是用UTF-8读啊。能否把乱码内容复制一段发出来？我可以试着看看。

另外你是第一次用Texifier，还是用了很久了呢？或者说，用Texifier打开其它项目正常吗？

**Comment by YDX-2147483647** [comment]
另外，请问你下载的是哪个模板？[Release v3.7.0 (Public Release) · BITNP/BIThesis](https://github.com/BITNP/BIThesis/releases/tag/v3.7.0) 的`paper-translation.zip`吗？

**Comment by Larerr** [comment]
你好！感谢您如此迅速的回复！我现在的情况是这样的：
1. 我将Texifier的编码格式改为了UTF-8（如下图所示），然后我重新打开软件之后，项目先是以中文短暂的显示了几秒后又变回了乱码。
<img width="1710" alt="image" src="https://github.com/BITNP/BIThesis/assets/101967188/22fda4de-d5a1-46dd-b179-79984e1e0c5b">

2. 别的项目是可以用滴～

3. 下载的模板是您所链接🔗的这个

4. 其中的一段乱码如下：
```
\BITSetup{
  cover = {
    % å¨å°é¢ä¸­è½½å¥æãåäº¬çå·¥å¤§å­¦ãå­æ ·çå¾çï¼å¦æ å¿è¦è¯·å¿æ¹å¨ã
    headerImage = images/header.png,
    % å¨å°é¢æ é¢ä¸­ä½¿ç¨ææºé»ä½ï¼ä½¿ç¨æ­¤éé¡¹å¯ä»¥ä¿è¯ä¸ Word å°é¢æ é¢çå­ä½ä¸è´ã
    xiheiFont = STXIHEI.TTF,
    % å®æ¹æ¨¡æ¿éç¨äºåºå®çä¸åçº¿å®½åº¦ãæä»¬éç¨ä»¥ä¸ä¸¤ä¸ªéé¡¹æ¥è¾¾æè¿ä¸ªææã
    % å¦æä½ æ³è¦ä½¿ç¨èªå¨è®¡ç®çä¸åçº¿å®½åº¦ï¼ä¹å¯ä»¥å å»ä»¥ä¸ä¸¤ä¸ªéé¡¹ã
    autoWidth = false,
    valueMaxWidth = 20em,
  },
  info = {
    title = åäº¬çå·¥å¤§å­¦æ¬ç§çæ¯ä¸è®¾è®¡ï¼è®ºæï¼é¢ç®,
    titleEn = {The Subject of Undergraduate Graduation Project (Thesis) of Beijing Institute of Technology},
    % æ³è¦å é¤æé¡¹å°é¢ä¿¡æ¯ï¼ç´æ¥å é¤è¯¥é¡¹å³å¯ã
    % æ³è¦è®©æé¡¹å°é¢ä¿¡æ¯çç©ºï¼ä½æ¯ä¿çä¸åçº¿ï¼ï¼è¯·ä¼ å¥ç©ºç½ç¬¦ç»æçå­ç¬¦ä¸²ï¼å¦"{~}"ã
    % å¦éè¦æ¢è¡ï¼åç¨ â\\â ç¬¦å·åå²ã
    school = è®¡ç®æºå­¦é¢,
    major = è®¡ç®æºç§å­¦ä¸ææ¯,
    author = æ è®¡ç®,
    class = 0596xxxx,
    studentId = 11xxxxxxxx,
    supervisor = å¼ åå¸,
    translationTitle = å¡«åä½ çå¤æç¿»è¯ä¸­æé¢ç®,
    translationOriginTitle = å¡«åä½ çå¤æç¿»è¯åè±æé¢ç®,
    keywords = {åäº¬çå·¥å¤§å­¦ï¼æ¬ç§çï¼æ¯ä¸è®¾è®¡ï¼å¤æç¿»è¯ï¼},
    % å¦æä½ çæ¯è®¾ä¸ºæ ¡å¤æ¯è®¾ï¼è¯·å°ä¸é¢è¿ä¸è¡è¯­å¥è§£é¤æ³¨éï¼å é¤ç¬¬ä¸ä¸ªç¾åå·å­ç¬¦ï¼å¹¶å¡«åä½ çæ ¡å¤æ¯è®¾å¯¼å¸åå­
    % externalSupervisor = å·¦åæ ,
  },
  style = {
    % head = {èªå®ä¹é¡µçæå­}
  }
}
```

**Comment by fky2015** [comment]
感觉是 Texifier 的问题：https://www.zhihu.com/question/47583918

这个我们除了为你提供一些建议，很难直接解决——因为我们模板本身是正常的。

**Comment by YDX-2147483647** [comment]
根据你提供的乱码，Texifiy是按Latin1编码读的。

![图片](https://github.com/BITNP/BIThesis/assets/73375426/7b6cb4bd-5f89-46a9-b39c-e1016f3b37f7)

> 先是以中文短暂的显示了几秒后又变回了乱码。

有可能软件有检测编码功能，错误地检测成了Latin1？

**Comment by Larerr** [comment]
感谢你们的回答，现在我已经解决了编码问题！但很遗憾还是不能正确显示内容，经过我的进一步调试，我貌似发现了问题其他存在原因。
这是BIT-Translate在Texifier中识别到的目录：
<img width="304" alt="image" src="https://github.com/BITNP/BIThesis/assets/101967188/d7991184-04e7-4712-8eb3-8d8ddb29a800">
这是ECCV2024在Texifier中识别到的目录：
<img width="333" alt="image" src="https://github.com/BITNP/BIThesis/assets/101967188/a6b67043-3862-47c4-bb7f-bac374322427">
我发现BIT模板被错误的识别了，”一级题目“”二级题目“等并不是main.tex中的分段，而是1_chapter.tex和其他文件中的设置。BIT和ECCV在Texifier中的打开方式是一致的，都是选取main.tex，并由Texifier对main.tex所处的当前文件夹目录进行自动识别并导入，但是貌似这步出现了错误。

**Comment by fky2015** [comment]
这个是编辑器实现的问题，只要不影响你论文写作的话，可以忽略。

**Comment by Larerr** [comment]
在疯狂的报错QAQ
<img width="1019" alt="image" src="https://github.com/BITNP/BIThesis/assets/101967188/bbcd0e8c-1183-4206-8a28-383e9d857341">

**Comment by Larerr** [comment]
我猜测有没有可能是文件夹识别的格式不同，导致很多路径的东西他找不到

**Comment by YDX-2147483647** [comment]
我看报的是字体问题，应该不是编辑器的事了。请问你知道自己用的 LaTeX 发行版是什么吗？

[🍌 如何开始 | BIThesis](https://bithesis.bitnp.net/guide/getting-started.html)

> 有没有可能是文件夹识别的格式不同

我估计不是，因为 LaTeX 并没有隐含规定文件夹，现在`chapters/`等文件夹都是像下面这样明面写出的——除非编译时的工作目录（current working directory）不对。

```latex
% main.tex
\input{chapters/1_chapter1.tex}
```
